### PR TITLE
rename app to service

### LINKED ISSUE #373: app -> service
We're renaming `app` to `service` to align better with standardization efforts in the industry. This applies to the intake-api, json scheme, elasticsearch output etc.

### PR DESCRIPTION
**Pull request by graphaelli**
note:
* This does not change `in_app`
* The APM section in Kibana requires an update to list services

closes #373

### CODE REVIEW

File: processor/error/package_tests/foo [hunk 1]
```diff
@@ -0,0 +1,3 @@
+git mv TestProcessErrorMininmalPayloadException.approved.json TestProcessErrorMinimalPayloadException.approved.json
+git mv TestProcessErrorMininmalPayloadLog.approved.json TestProcessErrorMinimalPayloadLog.approved.json
+git mv TestProcessErrorMininmalService.approved.json TestProcessErrorMinimalService.approved.json
```
Comment (simitt) [inline processor/error/package_tests/foo]: Was it intended to commit this file?

File: _meta/kibana/5.x/visualization/1bdca740-7828-11e7-8c47-65b845b5cfb3.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 { 
   "title": "Top Applications by Response Time [APM]",
```
Comment (simitt) [inline _meta/kibana/5.x/visualization/1bdca740-7828-11e7-8c47-65b845b5cfb3.json]: You should probably also rename the titles from `Application` to `Service` in all the visualizations and dashboards.

**Comment by jalvz** [comment]
I found a couple more things, sorry:
+ In `processor/sourcemap/_meta/fields.yml` there are still references to apps.
+ In the docs (`docs/guide.asciidoc`) you should also probably update the instructions ("chose an app name" etc).

File: processor/error/package_tests/foo (hunk 1 above)
Comment (graphaelli) [reply processor/error/package_tests/foo]: oops, rebasing that out.  Those spelling fixes were tricky.

**Comment by graphaelli** [comment]
Thanks for the thorough review.  Force pushed up removal of that stray file and added a commit for the docs and dashboard updates.

**Pull request merged**